File: app/src/main/java/cosmic/com/pkwprj/adapter/ProgressAdapter.java
package cosmic.com.pkwprj.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import cosmic.com.pkwprj.R;
import cosmic.com.pkwprj.model.Office;
import cosmic.com.pkwprj.view.SecondActivity;


public class ProgressAdapter extends RecyclerView.Adapter<ProgressAdapter.ViewHolder> {

    Context context;
    Office office;
    ArrayList<Office> lists;

    public ProgressAdapter(Context context, ArrayList<Office> list) {
        this.context = context;
        this.lists = list;

    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_office, parent, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        office = lists.get(position);

        holder.tv_officeName.setText(office.getName());
        holder.tv_location.setText(office.getLocation());
        holder.progressBar3.setProgressDrawable(office.getDrawable());

        if(0<SecondActivity.convertedKey&&SecondActivity.convertedKey<18) {
            int inputX = (SecondActivity.convertedKey - 1) * 50;
            holder.tv_current.setX(inputX);
        }else{
            holder.tv_current.setText("");
        }
    }


    @Override
    public int getItemCount() {
        return lists.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        ProgressBar progressBar3;
        TextView tv_officeName, tv_location,tv_current;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            progressBar3 = itemView.findViewById(R.id.progressBar3);
            tv_officeName = itemView.findViewById(R.id.tv_officeName);
            tv_location = itemView.findViewById(R.id.tv_officeLocation);
            tv_current=itemView.findViewById(R.id.tv_move);
        }

    }
}

File: app/src/main/java/cosmic/com/pkwprj/adapter/ProgressAdapter.kt
package cosmic.com.pkwprj.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ProgressBar
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import cosmic.com.pkwprj.R
import cosmic.com.pkwprj.model.Office
import cosmic.com.pkwprj.view.SecondActivity
import java.util.*


class ProgressAdapter(internal var context: Context, internal var lists: ArrayList<Office>) :
    RecyclerView.Adapter<ProgressAdapter.ViewHolder>() {
    internal lateinit var office: Office

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_office, parent, false)

        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        office = lists[position]

        holder.tv_officeName.text = office.name
        holder.tv_location.text = office.location
        holder.progressBar3.progressDrawable = office.drawable

        if (0 < SecondActivity.convertedKey && SecondActivity.convertedKey < 18) {
            val inputX = (SecondActivity.convertedKey - 1) * 50
            holder.tv_current.x = inputX.toFloat()
        } else {
            holder.tv_current.text = ""
        }
    }


    override fun getItemCount(): Int {
        return lists.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        internal var progressBar3: ProgressBar
        internal var tv_officeName: TextView
        internal var tv_location: TextView
        internal var tv_current: TextView

        init {

            progressBar3 = itemView.findViewById(R.id.progressBar3)
            tv_officeName = itemView.findViewById(R.id.tv_officeName)
            tv_location = itemView.findViewById(R.id.tv_officeLocation)
            tv_current = itemView.findViewById(R.id.tv_move)
        }

    }
}
